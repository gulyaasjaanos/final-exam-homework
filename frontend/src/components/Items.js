import React from 'react';
import { useEffect } from 'react';
import { Link } from 'react-router-dom';
import WithSession from './WithSession';
import { listitems } from '../actions';
import { connect } from 'react-redux';

function Items({items, listitems}) {

  useEffect( () => {  

    listitems()

  }, [listitems]);

  return (
    <>
      <p>ITEMS FOR SALE</p>
      { items.map( item => (
        <Link to={`/items/${item.id}`} key={item.id}>
          <ul className='itemCard' >
            {Object.keys(item).map( (key,i) => {
                return (key !== 'id') ? (<li key={i}>{item[key]}</li>) : null
              }
            )}
          </ul>
        </Link>
      ))}
    </>
  );
    
};

const mapStateToProps = state => (
  {
    items: state.items,
  }
);


const mapDispatchToProps = dispatch => (
  {
    listitems: () => dispatch( listitems() )
  }
);


export default connect(mapStateToProps,mapDispatchToProps)( WithSession(Items) );
