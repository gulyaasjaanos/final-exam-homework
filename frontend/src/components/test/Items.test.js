import React from 'react';
import { Provider } from 'react-redux';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import configureStore from 'redux-mock-store';
import { BrowserRouter,  Route } from 'react-router-dom';

import Items from '../Items';

let container;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('matches snapeshot', async () => {
  const mockStore = configureStore([]);
  const mockedStore = mockStore({
  });
  mockedStore.dispatch = jest.fn();
  await act(async () => {
    render(
      <Provider store={mockedStore}>
        <BrowserRouter>
          <Route>
            <Items />
          </Route>
        </BrowserRouter>
      </Provider>,
      container
    );
  });

  expect(container.innerHTML).toMatchSnapshot();
});

