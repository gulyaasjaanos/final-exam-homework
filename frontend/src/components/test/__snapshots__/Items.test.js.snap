// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapeshot 1`] = `""`;
